src/crawlee/request_loaders/_sitemap_request_loader.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
+
+import asyncio
+from logging import getLogger
+from typing import TYPE_CHECKING, Any
+
+from crawlee import Request
+from crawlee._utils.docs import docs_group
+from crawlee._utils.globs import Glob
+from crawlee._utils.sitemap import ParseSitemapOptions, SitemapSource, SitemapUrl, parse_sitemap
+from crawlee.request_loaders._request_loader import RequestLoader
+
+if TYPE_CHECKING:
+    import re
+    from collections.abc import Sequence
+
+    from crawlee.storage_clients.models import ProcessedRequest
+
+
+logger = getLogger(__name__)
+
+
+@docs_group('Classes')
+class SitemapRequestLoader(RequestLoader):
+    """A request loader that reads URLs from sitemap(s).
+
+    The loader fetches and parses sitemaps in the background, allowing crawling to start
+    before all URLs are loaded. It supports filtering URLs using glob and regex patterns.
+    """
+
+    def __init__(
+        self,
+        sitemap_urls: list[str],
+        *,
+        proxy_url: str | None = None,
```

Comment:
I'd prefer to accept an `HttpClient` instance here instead of just using `httpx`. I know that the Python `HttpClient` doesn't support streaming yet, but that's something we should fix anyway :grin:

Reply:
> So it's time to add a `stream` method 😄

It won't get easier later :slightly_smiling_face: If you can do it in a separate PR, please do.